### PR TITLE
 db integration for patch users route

### PR DESCRIPTION
**Pull request by stovalln21**
Database integration for patch users route

### CODE REVIEW

**Review by benrkaplan — APPROVED** [review]

**Review by sam-eng — CHANGES_REQUESTED** [review]
It looks good, but there's a fair amount that needs to be changed before this can get merged in

**Comment by sam-eng** [comment]
> It looks good, but there's a fair amount that needs to be changed before this can get merged in

I'll get to these in a bit, but please don't merge this in yet!

**Pull request closed without merging**
